src/main/java/gov/usgs/earthquake/geoserve/ANSSRegionsFactory.java

```diff
@@ -0,0 +1,197 @@
+
+    // global factory object
+    private static ANSSRegionsFactory SINGLETON;
+
+    // service used to load regions
+    private GeoserveLayersService geoserveLayersService;
+
+    // the current regions object.
+    private Regions regions;
+
+    // timer used to auto fetch region updates
+    private Timer updateTimer = new Timer();
+
+
+    /**
+     * Use default GeoserveLayersService.
+     */
+    public ANSSRegionsFactory () {
+        this(new GeoserveLayersService());
+    }
+
+    /**
+     * Use custom GeoserveLayersService.
+     */
+    public ANSSRegionsFactory (final GeoserveLayersService geoserveLayersService) {
+        this.geoserveLayersService = geoserveLayersService;
+    }
+
+    /**
+     * Get the global ANSSRegionsFactory, 
+     * creating and starting if needed.
+     */
+    public static ANSSRegionsFactory getFactory() {
+        if (SINGLETON == null) {
+            SINGLETON = new ANSSRegionsFactory();
+            SINGLETON.startup();
+            Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+                SINGLETON.shutdown();
+            }));
+        }
+        return SINGLETON;
+    }
+
+    /**
+     * Set the global ANSSRegionsFactory,
+     * shutting down any existing factory if needed.
+     */
+    public static void setFactory(final ANSSRegionsFactory factory) {
+        if (SINGLETON != null) {
+            SINGLETON.shutdown();
+        }
+        SINGLETON = factory;
+    }
+
+    /**
+     * Get the service.
+     */
+    public GeoserveLayersService getGeoserveLayersService() {
+        return this.geoserveLayersService;
+    }
+
+    /**
+     * Set the service.
+     */
+    public void setGeoserveLayersService(final GeoserveLayersService service) {
+        this.geoserveLayersService = service;
+    }
+
+    /**
+     * Get the most recently fetched Regions.
+     */
+    public Regions getRegions () {
+        return regions;
+    }
+
+    /**
+     * Download regions from geoserve.
+     *
+     * Writes out to "regions.json" in current working directory and,
+     * if unable to update, reads in local copy.
+     */
+    public void fetchRegions () {
+        LOGGER.info("Fetching ANSS Authoritative Regions from Geoserve");
+        File regionsJson = new File(REGIONS_JSON);
+
+        // try loading from geoserve
+        try {
+            JsonObject json = this.geoserveLayersService.getLayer("anss");
+            this.regions = new RegionsJSON().parseRegions(json);
+            LOGGER.info("Loaded ANSS Authoritative Regions from Geoserve");
+            // save regions if needed later
+            FileUtils.writeFileThenMove(
+                    new File(REGIONS_JSON + ".temp"),
+                    regionsJson,
+                    json.toString().getBytes());
+            LOGGER.finer("Storing ANSS Authoritative Regions to "
+                    + REGIONS_JSON);
+            // everything worked
+            return;
```

Comment:
Not a huge fan of these short-circuit returns. It makes the code harder to read/understand. Maybe two functions `loadFromGeoserve` and `loadFromFile`. Then one parent function that calls both as needed.

Reply:
Refactored into multiple functions